### PR TITLE
[WIP] glass: replace protractor with cypress

### PR DESCRIPTION
**Pull request by callithea**
Note: e2e tests will be added in separate PRs with dedicated github issues.
This PR only includes changes which are related to replacing the e2e framework by switching from protractor to cypress.

### CODE REVIEW

File: src/glass/cypress/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  "include": [
+    "**/*.ts",
+    "plugins/index.js"
+  ],
+  "compilerOptions": {
+    "sourceMap": false,
+    "types": [
+      "cypress"
+    ],
+    "target": "es6"
```
Comment (callithea) [inline src/glass/cypress/tsconfig.json:13]: I think I need to replace this with `es2015` or `2018`?

Comment (votdev) [reply src/glass/cypress/tsconfig.json:13]: I think we should remove the `target` property and inherit from the `tsconfig.json` (which is already done in line 2).

Comment (votdev) [reply src/glass/cypress/tsconfig.json:13]: > Guess I also need to add an "exclude 'cypress'" to glass/tsconfig.json.

Yes, do that please.

**Comment by votdev** [comment]
I'm not able to run the tests. Could you please include some documentation that explains how to install and use the E2E environment locally.
```
$ npm install
$ npm run e2e:ci                                                   

> glass@0.0.0 e2e:ci
> start-test 4200 'cypress run -b chrome --headless'

sh: 1: start-test: not found
```

**Comment by callithea** [comment]
> I'm not able to run the tests. Could you please include some documentation that explains how to install and use the E2E environment locally.
> 
> ```
> $ npm install
> $ npm run e2e:ci                                                   
> 
> > glass@0.0.0 e2e:ci
> > start-test 4200 'cypress run -b chrome --headless'
> 
> sh: 1: start-test: not found
> ```

Thanks for testing! Unfortunately I forgot to add an npm package. Hope it works now.

File: src/glass/package.json [hunk 2]
```diff
@@ -52,10 +53,12 @@
     "@mdi/angular-material": "5.9.55",
     "@ngx-translate/core": "^13.0.0",
     "@swimlane/ngx-charts": "^17.0.1",
+    "cypress": "^7.1.0",
```
Comment (kshtsk) [inline src/glass/package.json:56]: is this the right place for it? Maybe the dev dependency is better?

**Comment by kshtsk** [comment]
rebase required

File: src/glass/package.json (hunk 2 above)
Comment (callithea) [reply src/glass/package.json:57]: Ack, I'll move it to `devDependencies` :+1:

File: src/glass/package.json [hunk 3]
```diff
@@ -11,10 +11,11 @@
     "lint:eslint": "ng lint",
     "lint:html": "htmllint src/app/**/*.html && html-linter --config html-linter.config.json",
     "lint": "run-p -csl --aggregate-output lint:*",
-    "fix:prettier": "prettier --write \"{src,e2e}/**/*.{ts,scss}\"",
+    "fix:prettier": "prettier --write \"{src,cypress}/**/*.{ts,scss}\"",
     "fix:eslint": "npm run lint:eslint -- --fix",
     "fix": "run-p -csl --aggregate-output fix:*",
-    "e2e": "ng e2e",
+    "e2e": "start-test 4200 'cypress open'",
```
Comment (votdev) [inline src/glass/package.json:17]:
```suggestion
    "e2e": "start-test 4242 'cypress open'",
```

File: src/glass/package.json [hunk 4]
```diff
@@ -11,10 +11,11 @@
...
+    "e2e:ci": "start-test 4200 'cypress run -b chrome --headless'",
```
Comment (votdev) [inline src/glass/package.json:18]:
```suggestion
    "e2e:ci": "start-test 4242 'cypress run -b chrome --headless'",
```

**Review by votdev — CHANGES_REQUESTED** [review]

**Comment by kshtsk** [comment]
so.... what is the status? do we expect this is working at some point in the nearest future?

**Comment by callithea** [comment]
Closing this PR in favor of #584. Thank you for taking over, @kshtsk!

**Pull request closed without merging**